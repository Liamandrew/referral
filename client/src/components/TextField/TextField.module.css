.input {
    border: 0;
    width: 100%;
    height: 48px;
    font-size: 18px;
    box-sizing: border-box;
    padding: 0 12px;
}

.inputContainer {
    position: relative;
    border: 2px solid #bebebe;
    border-radius: 2px;
}

.label {
    display: block;
    font-size: 12px;
    color: #918f8f;
    text-transform: uppercase;
    margin-bottom: 4px;
}

.error {
    color: red;
    font-size: 12px;
    height: 14px;
    padding-top: 4px;
    padding-left: 4px;
    margin-block-start: unset;
    margin-block-end: unset;
}
