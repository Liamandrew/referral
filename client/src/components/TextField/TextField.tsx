import React from "react";
import style from "./TextField.module.css";

interface TextFieldProps {
    label: string;
    name?: string;
    error?: string;
}

const TextField = React.forwardRef<HTMLInputElement, TextFieldProps>(
    ({ name, label, error }, ref) => (
        <div>
            <label id={name} className={style.label}>
                {label}
            </label>
            <div className={style.inputContainer}>
                <input
                    ref={ref}
                    className={style.input}
                    name={name}
                    aria-labelledby={name}
                />
            </div>
            {error && <p className={style.error}>{error}</p>}
        </div>
    )
);

export { TextField };
