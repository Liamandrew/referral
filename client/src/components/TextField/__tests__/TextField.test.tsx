import React from "react";
import { render } from "@testing-library/react";
import { TextField } from "../TextField";

describe("<TextField>", () => {
    test("should match snapshot in default state", () => {
        const { container } = render(<TextField label="label" />);

        expect(container).toMatchSnapshot();
    });

    test("should match snapshot in default state", () => {
        const { container, getByText } = render(
            <TextField label="label" error="there is an error" />
        );

        expect(getByText(/there is an error/i)).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });
});
