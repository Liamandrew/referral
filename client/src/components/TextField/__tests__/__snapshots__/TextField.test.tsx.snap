// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField> should match snapshot in default state 1`] = `
<div>
  <div>
    <label
      class="label"
    >
      label
    </label>
    <div
      class="inputContainer"
    >
      <input
        class="input"
      />
    </div>
  </div>
</div>
`;

exports[`<TextField> should match snapshot in default state 2`] = `
<div>
  <div>
    <label
      class="label"
    >
      label
    </label>
    <div
      class="inputContainer"
    >
      <input
        class="input"
      />
    </div>
    <p
      class="error"
    >
      there is an error
    </p>
  </div>
</div>
`;
